common/membership/ringpop/fx.go

```diff
@@ -25,71 +25,36 @@
 package ringpop
 
 import (
-	"context"
-
 	"go.uber.org/fx"
 
-	"go.temporal.io/server/common/config"
-	"go.temporal.io/server/common/dynamicconfig"
-	"go.temporal.io/server/common/log"
 	"go.temporal.io/server/common/membership"
-	"go.temporal.io/server/common/persistence"
-	"go.temporal.io/server/common/primitives"
-	"go.temporal.io/server/common/rpc/encryption"
 )
 
 var Module = fx.Options(
-	fx.Provide(membershipMonitorProvider),
+	fx.Provide(
+		provideFactory,
+		provideMonitor,
```

Comment:
if we have fx to wire params through and control initialization order and lifecycle, why are we still doing this two-stage factory→monitor thing? can we collapse it into one?

Reply:
We can. I want to get rid of the factory object later and just let fx and monitor do all of the initialization. That would also help us get rid of the logger.Fatal statements as well; IMO, these should be replaced with errors provided by `pkg/errors` so that we can still get the whole stack trace without forgoing the additional error info that fx provides.

Reply:
I removed provideFactory though and merged it with provideMonitor